src/mixins/filing-template-mixin.ts

```diff
@@ -560,21 +569,28 @@ export default class FilingTemplateMixin extends Mixins(DateMixin) {
 
     // Restore business information
     this.setBusinessInformation({
-      ...filing.business,
-      ...filing.changeOfRegistration.business
+      ...entitySnapshot.businessInfo
     })
 
+    // Restore Naics
+    if (filing.changeOfRegistration.business.naics) {
+      this.setNaics(filing.changeOfRegistration.business.naics)
+    }
+
     // Restore name request
     this.setNameRequest(filing.changeOfRegistration.nameRequest || { legalName: entitySnapshot.businessInfo.legalName })
 
     // Store office addresses
-    this.setOfficeAddresses({
-      registeredOffice: {
-        mailingAddress: filing.changeOfRegistration.businessAddress.mailingAddress,
-        deliveryAddress: filing.changeOfRegistration.businessAddress.deliveryAddress
+    let addresses
+    if (filing.changeOfRegistration.businessAddress) {
+      addresses = {
+        registeredOffice: {
+          mailingAddress: filing.changeOfRegistration.businessAddress.mailingAddress,
+          deliveryAddress: filing.changeOfRegistration.businessAddress.deliveryAddress
+        }
```

Comment:
`registeredOffice: filing.changeOfRegistration.businessAddress` ?

Reply:
Sorry, I wasn't clear all right...

I'm wondering why you de-reference filing.changeOfRegistration.businessAddress and objectify the props, instead of just assigning it directly as in my example?

Reply:
Oh i see. 
hahah thank you for the clarification, you are right. I missed that!

Reply:
Updated